common/src/main/java/com/tc/logging/TCLogging.java

```diff
@@ -344,154 +346,101 @@ public static void setLogDirectory(File theDirectory, int processType) {
         try {
-          if (theDirectory.getCanonicalPath().equals(currentLoggingDirectory.getCanonicalPath())) { return; }
+          if (theDirectory.getCanonicalPath().equals(currentLoggingDirectory.getCanonicalPath())) {
+            return;
+          }
         } catch (IOException ioe) {
           // oh, well -- what can we do? we'll continue on.
         }
       }
     }
 
-    try {
-      FileUtils.forceMkdir(theDirectory);
-    } catch (IOException ioe) {
-      reportLoggingError("We can't create the directory '" + theDirectory.getAbsolutePath()
-                         + "' that you specified for your logs.", ioe);
-      return;
-    }
-
-    if (!theDirectory.canWrite()) {
-      // formatting
-      reportLoggingError("The log directory, '" + theDirectory.getAbsolutePath() + "', can't be written to.", null);
+    if (!createAndValidateLogDir(theDirectory)) {
       return;
     }
 
     FileLock thisDirectoryLock = null;
+    String lockFileName;
+    int counter = 0;
+    boolean isLockAcquired = false;
+    String baseName;
 
-    if (!lockingDisabled) {
-      File lockFile = new File(theDirectory, LOCK_FILE_NAME);
+    switch (processType) {
+      case PROCESS_TYPE_L2:
+      case PROCESS_TYPE_GENERIC:
+        baseName = "terracotta-logging";
+        break;
+      case PROCESS_TYPE_L1:
+        baseName = clientName;
+        break;
+      default:
+        throw Assert.failure("Unknown process type: " + processType);
+    }
 
-      try {
-        lockFile.createNewFile();
-        Assert.eval(lockFile.exists());
-        FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
-        thisDirectoryLock = channel.tryLock();
-
-        if (thisDirectoryLock == null) {
-          reportLoggingError("The log directory, '" + theDirectory.getAbsolutePath()
-                             + "', is already in use by another "
-                             + "Terracotta process. Logging will proceed to the console only.", null);
+    if (!lockingDisabled) {
+      while (!isLockAcquired && counter < 10) {
+        if (counter == 0) {
+          lockFileName = "." + baseName + ".lock";
+        } else {
+          lockFileName = "." + baseName + ".collision-" + counter + ".lock";
+        }
+        File lockFile = new File(theDirectory, lockFileName);
+        try {
+          lockFile.createNewFile();
+          Assert.eval(lockFile.exists());
+          FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
+          thisDirectoryLock = channel.tryLock();
+
+          if (thisDirectoryLock == null) {
+            // try different lock
+            counter++;
+            continue;
+          }
+        } catch (OverlappingFileLockException ofle) {
+          // try different lock file name
+          counter++;
+          continue;
+        } catch (IOException ioe) {
+          reportLoggingError("We can't lock the file '" + lockFile.getAbsolutePath() + "', to make sure that only one "
```

Comment:
I am generally not a fan of personalizing log messages using "we".  I suggest alternate wording such as. ."The file XXX cannot be locked..."

Reply:
Does anything remove these lock files?

Reply:
these lock files are not removed

Reply:
these lock files are not removed

> I am generally not a fan of personalizing log messages using "we". I suggest alternate wording such as. ."The file XXX cannot be locked..."

these are existing messages